core/baseline-core-v1.0-psd01.md

```diff
@@ -4048,60 +4050,229 @@ This section of the document focuses on the requirements and considerations rela
+
+Test Steps:
+
+1. A BPI workgroup participant initiates an agreement process among the BPI subjects in the workgroup.
+2. The BPI system presents the BPI subjects with the source and type of the external authoritative data.
+3. The BPI subjects review and validate the source and type of the external authoritative data.
+4. If the specified quorum of BPI subjects agree, the BPI system records the agreement and uses the external authoritative data as the input source for the workstep.
+5. If the quorum of BPI subject is not reached, the BPI system does not record the agreement and prompts the subjects to discuss and resolve the disagreement.
+6. Once the agreement has been reached, the BPI system records the agreement and uses the external authoritative data as the input source for the workstep.
+7. Start the BPI workstep that requires external authoritative data as input.
+8. Complete the workstep using the agreed-upon external authoritative data.
+9. Verify that the workstep was successfully completed using the agreed upon external authoritative data.
+
+Test Passing Criteria: The test is considered passing if,
+
+* The BPI system successfully initiates the agreement process among the BPI subjects.
+* The specified quorum of BPI subjects agree upon the source and type of the external authoritative data.
+    * The BPI system records the agreement and uses the agreed-upon external authoritative data as the input source for the workstep.
+    * The workstep is successfully completed using the agreed-upon external authoritative data.
+* The BPI system responds as configured if the specified quorum of BPI subjects on the source and type of the external authoritative data is not reached.
 
 ## 8.3 External Non-authoritative, Non-deterministic Data for BPIs
 
+###### extnonauthbpidata
+
 External non-authoritative, non-deterministic data means that there does not exist an authoritative, deterministic source for the data used as input to a BPI workstep. In that case, a normalizing mechanism often referred to as an oracle is needed to account for potential source discrepancies. Entities reporting external non-authoritative, non-deterministic data into a BPI worksteps may also be referred to as oracles.
 
 ### 8.3.1 Data Trustworthiness
 
+###### bpidatatrust
+
 External non-authoritative, non-deterministic data for BPIs resides outside of an authoritative source and is subject to manipulation. As such, steps should be taken to remove counterparty manipulation and error risk through mechanisms such as redundancy in data reporting and error checking, either cross-party, or by an appropriately incentivized 3rd party.
 
 #### **[D42]** 
-External non-authoritative, non-deterministic BPI input data into a BPI workstep SHOULD be sourced from multiple endpoints.
+External non-authoritative, non-deterministic BPI input data into a BPI workstep SHOULD be sourced from multiple sources.
+
```

Comment:
given the existence of R316 this requirement seems completely unnecessary. I suggest removing the 22 lines from the spec, as an improvement.

Reply:
R316 and D42 are not mutually exclusive. R316 is a validation requirement of the data ... from one or more sources. D42 says that in this scenario the same data should be coming from multiple sources for cross-validation purposes. Does that make sense?

Reply:
It's not that they are exclusive, but they're redundant. If you are validating the data already, do we also want to say that you should have to cross-validate it by getting it from multiple sources?

(It is possible that the answer is "yes we do because we're prepared to make people do extra work for extra security")

Reply:
> It's not that they are exclusive, but they're redundant. If you are validating the data already, do we also want to say that you should have to cross-validate it by getting it from multiple sources?
> 
> (It is possible that the answer is "yes we do because we're prepared to make people do extra work for extra security")

Yes, that is the idea behind D42. This is to enhance security.